### ISSUE TITLE
nvcc compile error: attribute does not apply to any entity

### ISSUE THREAD

**Issue by p1m3nt** [gabime/spdlog#1261]
When I tried to use spdlog in a .cu file, I got following errors:

`1>c:\spdlog-1.x\include\spdlog\fmt\bundled/core.h(477): error : attribute does not apply to any entity`
`1>c:\spdlog-1.x\include\spdlog\fmt\bundled/format.h(2599): error : attribute does not apply to any entity`
`1>C:\Program Files (x86)\Microsoft Visual Studio\2017\Professional\Common7\IDE\VC\VCTargets\BuildCustomizations\CUDA 10.0.targets(712,9): error MSB3721: The command ""C:\Program Files\NVIDIA GPU Computing Toolkit\CUDA\v10.0\bin\nvcc.exe" -gencode=arch=compute_61,code=\"sm_61,compute_61\" --use-local-env -ccbin "C:\Program Files (x86)\Microsoft Visual Studio\2017\Professional\VC\Tools\MSVC\14.16.27023\bin\HostX86\x64" -x cu -I"C:\spdlog-1.x\include" -I"C:\Program Files\NVIDIA GPU Computing Toolkit\CUDA\v10.0\include" -I"C:\Program Files\NVIDIA GPU Computing Toolkit\CUDA\v10.0\include"  -G   --keep-dir x64\Debug -maxrregcount=0  --machine 64 --compile -cudart static  -g   -DWIN32 -DWIN64 -D_DEBUG -D_CONSOLE -D_MBCS -Xcompiler "/EHsc /W3 /nologo /Od /FdC:\test\bin\intermediates\x64\Debug\test\vc141.pdb /FS /Zi /RTC1 /MDd " -o C:\test\bin\intermediates\x64\Debug\test\kernel1.cu.obj "C:\test\test\kernel1.cu"" exited with code 1.`
`1>Done building project "test.vcxproj" -- FAILED.`

I searched around found that [nvcc does not support [[deprecated]] on windows](https://github.com/pytorch/pytorch/commit/5b33698776b468e212990e00eb0b6f8fc145ea99). 
I also read the related [nvcc issues](https://github.com/gabime/spdlog/issues?utf8=✓&q=nvcc), which I'm not sure if they are relevant to the errors.
Any seggestions how to fix it? Thanks.

env:
Win 10
VS2017(platform toolset v141) with SDK ver 10.0.17763.0
nvcc release 10.0, V10.0.130

**Comment by gabime** [comment]
core.h is part of fmt repo. Better to ask there.

**Comment by p1m3nt** [comment]
> core.h is part of fmt repo. Better to ask there.

Thanks @gabime for your suggestion. 

I've checked fmt repo and it seems [they've addressed the issue](https://github.com/fmtlib/fmt/commit/744302add085f30120c57f73cddbcb826cc92de9) recently. I am also able to build and run the tests of fmt successfully, especially the [fmt-in-cuda-test](https://github.com/fmtlib/fmt/tree/master/test/cuda-test) project.

I've also tried to replace the files in fmt bundled folder of spdlog with fmt's master branch, the errors persist. Thank you very much.
